pkg/kapp/app/interfaces.go

```diff
@@ -17,6 +17,8 @@ type App interface {
 	LabelSelector() (labels.Selector, error)
 	UsedGVs() ([]schema.GroupVersion, error)
 	UpdateUsedGVs([]schema.GroupVersion) error
+	UsedGKs() ([]schema.GroupKind, error)
+	UpdateUsedGKs([]schema.GroupKind, bool) ([]schema.GroupKind, error)
```

Comment:
`app.UsedGKs()` returns values from the cached app meta which is not updated. And we need to account for new KGs added in the manifest. Will explore if we can update the cache here as well. Not sure about it though.

This is so because so far:
1. All updates to the app meta were meant to be cached for the next deploy
2. The values used for the current deploy was the cached meta fetched from the config map.

Reply:
Updating cache, and using UsedGVKs now 🚀